src/snowflake/cli/_plugins/snowpark/snowpark_project_paths.py

```diff
@@ -56,18 +72,101 @@ def snowflake_requirements(self) -> SecurePath:
     def requirements(self) -> SecurePath:
         return SecurePath(self.path_relative_to_root(Path("requirements.txt")))
 
+
+@dataclass(unsafe_hash=True)
+class Artefact:
+    """Helper for getting paths related to given artefact."""
+
+    project_root: Path
+    path: Path
+    dest: str | None = None
+
+    def __init__(
+        self, project_root: Path, path: Path, dest: Optional[str] = None
+    ) -> None:
+        self.project_root = project_root
+        self.path = path
+        self.dest = dest
+        if self.dest and not self._is_dest_a_file() and not self.dest.endswith("/"):
+            self.dest = self.dest + "/"
+
+    @property
+    def _artefact_name(self) -> str:
+        if "*" in str(self.path):
+            before_wildcard = str(self.path).split("*")[0]
+            last_part = Path(before_wildcard).absolute().parts[-1]
+            return last_part + ".zip"
+        if (self.project_root / self.path).is_dir():
+            return self.path.stem + ".zip"
+        if (self.project_root / self.path).is_file() and self._is_dest_a_file():
+            return Path(self.dest).name  # type: ignore
+        return self.path.name
+
     @property
-    def dependencies(self) -> Path:
-        return self.path_relative_to_root(Path("dependencies.zip"))
+    def post_build_path(self) -> Path:
+        """
+        Returns post-build artefact path. Directories are mapped to corresponding .zip files.
+        """
+        deploy_root = self.deploy_root()
+        path = self._path_until_asterix() if "*" in str(self.path) else self.path.parent
+        if self._is_dest_a_file():
+            return deploy_root / self.dest  # type: ignore
+        return deploy_root / (self.dest or path) / self._artefact_name
+
+    def upload_path(self, stage: FQN | str | None) -> str:
+        """
+        Path on stage to which the artefact should be uploaded.
+        """
+        stage = stage or DEPLOYMENT_STAGE
+        if isinstance(stage, str):
+            stage = FQN.from_stage(stage).using_context()
+
+        stage_path = PurePosixPath(f"@{stage}")
+        if self.dest:
+            stage_path /= (
+                PurePosixPath(self.dest).parent if self._is_dest_a_file() else self.dest
+            )
+        else:
+            stage_path /= (
+                self._path_until_asterix()
+                if "*" in str(self.path)
+                else PurePosixPath(self.path).parent
+            )
+
+        return str(stage_path) + "/"
+
+    def import_path(self, stage: FQN | str | None) -> str:
+        """Path for UDF/sproc imports clause."""
+        return self.upload_path(stage) + self._artefact_name
+
+    def deploy_root(self) -> Path:
+        return self.project_root / "output"
+
+    def _is_dest_a_file(self) -> bool:
+        if not self.dest:
+            return False
+        return re.search(r"\.[a-zA-Z0-9]{2,4}$", self.dest) is not None
+
+    def _path_until_asterix(self) -> Path:
```

Comment:
nit: it's _ asterisk_. Asterix, AFAIK, is only the comic book character :)

Reply:
Changed